lib/compress.js

```diff
@@ -4321,6 +4332,20 @@ merge(Compressor.prototype, {
                 })
             }).optimize(compressor);
         }
+        if (compressor.option("properties") && self.expression instanceof AST_Object) {
```

Comment:
Done in https://github.com/mishoo/UglifyJS2/pull/2209/files#diff-230c53e13e3ee367aec7d49470eb42caR4335